Support compiling on msys2/ucrt64 (no exception handling)

qmfrederik opened the pull request:
This contains a minimal set of changes from #190 to get libobjc2 to compile under msys2/ucrt64. Exception handling is broken, but it at least allows for some kind of testing with libobjc2 on that environment.

I had to configure the project with `CC=clang CXX=clang LDFLAGS="-fuse-ld=lld -lstdc++ -lgcc_s" cmake .. -DTESTS=ON`.
79% tests passed, 40 tests failed out of 194.

I hope these changes are non-controversial and getting them merged into master prevent them from bit rotting on a branch (like #190) - further work could come as smaller, follow-up PRs.

Any objections?

qmfrederik commented on objc_msgSend.x86-32.S:
```diff
@@ -126,7 +126,13 @@ CDECL(objc_msgSend_stret):
         .ascii  " /EXPORT:_objc_msgSend_stret"
         .ascii  " /EXPORT:_objc_msgSend_fpret"
+#else
+        .ascii  " /EXPORT:objc_msgSend"
```
I'm not 100% sure about using the existing macros, but I added a new macro to `common.S` which simplified things. Thanks!

davidchisnall commented on objc_msgSend.x86-32.S:
```diff
@@ -37,7 +37,7 @@
 	jz    5f                             # Nil slot - invoke some kind of forwarding mechanism
 	mov   SLOT_OFFSET(%eax), %ecx
-#ifdef _WIN32
+#ifdef _MSC_VER
```
Does MSYS not support CFG?

qmfrederik replied:
It looks like this recently landed: https://sourceforge.net/p/mingw-w64/mailman/message/37712869/, I’ll have a closer look later today/tomorrow

qmfrederik replied:
mingw does support CFG, I've reverted this change.

davidchisnall commented:
Looks good to me. Please can you rebase and squash?

qmfrederik commented:
Thanks, @davidchisnall .  I rebased and squashed the commits.  Looks like the FreeBSD build failures on this PR and #255 are unrelated:

```
pkg install -y cmake ninja git
Updating FreeBSD repository catalogue...
pkg: http://pkgmir.geo.freebsd.org/FreeBSD:12:amd64/quarterly/meta.txz: Not Found
repository FreeBSD has no meta file, using default settings
pkg: http://pkgmir.geo.freebsd.org/FreeBSD:12:amd64/quarterly/packagesite.pkg: Not Found
pkg: http://pkgmir.geo.freebsd.org/FreeBSD:12:amd64/quarterly/packagesite.txz: Not Found
Unable to update repository FreeBSD
Error updating repositories!
```

qmfrederik commented:
Rebased on master
